Toolkit/ArcGISToolkit/TemplatePickerViewController.swift

```diff
@@ -0,0 +1,296 @@
+                
+                let infos = tableTemplates + typeTemplates
+                
+                // add to list of unfiltered infos
+                strongSelf.unfilteredInfos.append(contentsOf: infos)
+                
+                // re-assign to the current infos so we can update the tableview
+                // only should do this if not currently filtering
+                if !strongSelf.isFiltering{
+                    strongSelf.currentInfos = strongSelf.unfilteredInfos
+                }
+                
+                // generate swatches for the layer infos
+                for (index, info) in infos.enumerated(){
+                    if let feature = info.featureTable.createFeature(with: info.featureTemplate){
+                        let sym = info.featureLayer.renderer?.symbol(for: feature)
+                        sym?.createSwatch{ image, error in
+                            
+                            if error != nil{
+                                return
+                            }
+                            
+                            // update info with swatch
+                            infos[index].swatch = image
+                            
+                            // reload index where that info currently is
+                            if let index = strongSelf.indexPathForInfo(info){
+                                strongSelf.tableView.reloadRows(at: [index], with: .automatic)
+                            }
+                        }
+                    }
+                }
+            }
+            
+        }
+        
+    }
+    
+    // MARK: TableView delegate/datasource methods
+    
+    public func numberOfSectionsInTableView(_ tableView: UITableView) -> Int{
+        return tables.count
+    }
+    
+    public func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        return tables.count == 0 ? nil : tables[section].tableName
+    }
+    
+    public func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        
+        // when the user taps on a feature type
+        
+        // first get the selected object
+        let selectedFeatureTemplateInfo = infoForIndexPath(indexPath)!
+        
+        // If the search controller is still active, the delegate will not be
+        // able to dismiss us, if desired.
+        // Note: we can't do this before the above, or else the object we pull
+        // out of the datasource will be incorrect
+        navigationItem.searchController?.isActive = false
+        
+        delegate?.templatePickerViewControllerDidSelectTemplate(self, featureTemplateInfo: selectedFeatureTemplateInfo)
+    }
+    
+    override public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        
+        guard tables.count > 0 else {
```

Comment:
Again, the table view won't ask you for the number of rows in any section unless you tell it there are more than zero sections, so this check is unneeded.

Reply:
ok thank you